packages/editor-ui/src/stores/workflowHistory.store.ts

```diff
@@ -40,9 +45,65 @@ export const useWorkflowHistoryStore = defineStore('workflowHistory', () => {
 			return null;
 		});
 
+	const downloadVersion = async (workflowId: string, workflowVersionId: WorkflowVersionId) => {
+		const [workflow, workflowVersion] = await Promise.all([
+			workflowsStore.fetchWorkflow(workflowId),
+			getWorkflowVersion(workflowId, workflowVersionId),
+		]);
+		if (workflowVersion?.nodes && workflowVersion?.connections && workflow) {
```

Comment:
Why are we checking for the presence of connections? What if the workflow has nodes but no connections? Same applies below.

Reply:
On that note, what do we want to occur if the user happens to have a workflow with no nodes? I'd expect this empty workflow should be treated the same as a normal workflow.

Reply:
Makes sense, removed the conditions